pkg/scheduler/core/extender.go

```diff
@@ -299,8 +301,14 @@ func (h *HTTPExtender) Filter(
 
 	if h.nodeCacheCapable && result.NodeNames != nil {
 		nodeResult = make([]*v1.Node, 0, len(*result.NodeNames))
-		for i := range *result.NodeNames {
-			nodeResult = append(nodeResult, nodeNameToInfo[(*result.NodeNames)[i]].Node())
+		for _, nodeName := range *result.NodeNames {
```

Comment:
@goodluckbot first thanks review :), let me explain one by one:

1. In the first stage, I think we can just return an error, and expect the inconsistency should can be reconciled by the extender.

- It is already much better than crash itself, then backoff retry which hurt the whole cluster.

- Besides, the rest filttered nodes may be also inconsistent (outdated compared with default scheduler), so just return an error is the most safe way.

- And even if we just return the rest, just logging is not enough, we need to tell outside (end user and extender) that "the default scheduler does not fully respect your descion, please aware."

   So, it seems need more things to do to be perfect, I perfer keep **improving** this in future PR, instead of inside this **hotfix** PR.

2. Basically, we should not make assumption on what result extender will return, because extender is not offical k8s code. And one k8s can run multiple extenders plus default scheduler, if one extender behaves badly (such as its code has some bugs), k8s should just ignore it, instead of crash the whole k8s itself.